share/completions/clasp.fish

```diff
@@ -1,104 +1,171 @@
+function __fish_clasp_list_projects
+    clasp list --noShorten true 2>/dev/null |
+        string replace --regex '(.*) - https://script.google.com/d/(.*)/edit' '$2\\t$1'
+end
+
+function __fish_clasp_list_versions
+    clasp versions 2>/dev/null |
+        string replace --regex '(\\d+) - (.*)' '$1\\t$2' |
+        sed -n '2,$p'
+end
+
+function __fish_clasp_list_deployments
+    clasp deployments 2>/dev/null |
+        string replace --regex -- '- (\\S+) @(\\S+).*' '$1\\tversion $2' |
+        sed -n '2,$p'
+end
+
+function __fish_list_advanced_services
+    printf '%s\\t%s service\n' admin-sdk-directory 'Admin SDK Directory' \
+        admin-sdk-license-manager 'Admin SDK Enterprise License Manager' \
+        admin-sdk-groups-migration 'Admin SDK Groups Migration' \
+        admin-sdk-groups-settings 'Admin SDK Groups Settings' \
+        admin-sdk-reseller 'Admin SDK Google Workspace Reseller' \
+        admin-sdk-reports 'Admin SDK Reports' \
+        calendar 'Advanced Calendar' \
+        chat 'Advanced Chat' \
+        docs 'Advanced Docs' \
+        drive 'Advanced Drive' \
+        drive-activity 'Google Drive Activity' \
+        drive-labels 'Advanced Drive Labels' \
+        gmail 'Advanced Gmail' \
+        sheets 'Advanced Sheets' \
+        slides 'Advanced Slides' \
+        classroom Classroom \
+        people 'Advanced People' \
+        contacts-people 'Migrate from Contacts service to People API' \
+        tasks Tasks
+end
+
+function __fish_clasp_list_functions
+    grep --extended-regexp '^\\s*function\\s+\\w+' --recursive --include '*.js' --no-filename --color=never . |
+        string replace --regex '\\s*function\\s+(\\w+).*' '$1'
+end
+
+function __fish_clasp_list_subcommands
+    clasp --help |
+        string match --regex '^  [a-z]' --entire |
+        string replace --regex '^\\s{2}([a-z]+).*' '$1'
+end
+
+function __fish_seen_subcommands_from
```

Comment:
`__fish_seen_subcommands_from` has a name clash.
If the generic one doesn't work for some reason, it's fine to pu it into `clasp.fish` but use the `__fish_clasp` prefix if you do that.
Else things will behave differently depending on which completion script happens to be loaded first.